src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/SearchParameters/SearchParameterToTypeResolverTests.cs

```diff
@@ -18,22 +18,51 @@ public class SearchParameterToTypeResolverTests
         private static readonly FhirPathCompiler _compiler = new FhirPathCompiler();
 
         [Fact]
-        public void GivenAFhirPathExpressionWithFirstFunction_WhenResolvingTypes_ThenTheyAreRetunedCorrectly()
+        public void GivenAFhirPathExpressionWithFirstFunction_WhenResolvingTypes_ThenTheyAreReturnedCorrectly()
         {
             var expression = _compiler.Parse("Patient.extension.where(url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-race').first().extension.where(url = 'ombCategory').value");
             SearchParameterTypeResult[] results = SearchParameterToTypeResolver.Resolve(
                 KnownResourceTypes.Patient,
                 (SearchParamType.Token, expression, new Uri("http://hl7.org/fhir/SearchParameter/Patient-race")),
                 null).ToArray();
+            var types = results.Select(x => x.FhirNodeType).ToArray();
 
             if (ModelInfoProvider.Version == FhirSpecification.Stu3)
             {
-                Assert.Equal(38, results?.Length);
+                Assert.Equal(51, types.Length);
+            }
+            else if (ModelInfoProvider.Version == FhirSpecification.R4)
+            {
+                Assert.Equal(59, types.Length);
```

Comment:
Same with R4, should this be 49? What are these extra datatypes?

Reply:
before:
https://github.com/FirelyTeam/firely-net-sdk/blob/f028de2ea5d5b937f4a51f79c1dcbe9887615204/src/Hl7.Fhir.Core/Model/Extension.cs#L87

now:
https://github.com/FirelyTeam/firely-net-common/blob/127c771950e2a5f852607d0b3699cce894ea2126/src/Hl7.Fhir.Support.Poco/Model/Extension.cs#L82

If previously they stated which parameters are ok, now I need to improvise and derive them on a fly.

So I do this:
```
foreach (var pair in ModelInfo.FhirTypeToCsType)
                                {
                                    if ((ModelInfo.IsDataType(pair.Value) || ModelInfo.IsPrimitive(pair.Value))
                                        && !pair.Value.IsAbstract)
```

https://www.hl7.org/fhir/datatypes.html
In the past they also derived General-Purpose Data type from DataType, and rest (with exception of primitive) was derived from element.
Not the case anymore, everything is derived from DataType, so metadata types get leaked into it, special purpose types.
I don't see way to filter them out automatically, and manually doesn't feel right.

![image](https://user-images.githubusercontent.com/1523833/116489304-886f3300-a849-11eb-9f97-37ec81985960.png)